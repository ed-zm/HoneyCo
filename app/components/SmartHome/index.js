import React from 'react'
import { Row, Container } from 'reactstrap'
import styled from 'styled-components'
import { HoneyCoDarkGray } from '../../constants'
import SmartHomeBackground from './static/smart-home.jpg'
import Icon1 from './static/icons/icon1.png'
import Icon2 from './static/icons/icon2.png'
import Icon3 from './static/icons/icon3.png'
import Icon4 from './static/icons/icon4.png'
import Icon5 from './static/icons/icon5.png'

const StyledRow = styled(Row)`
  height: 768px;
  background: url(${SmartHomeBackground});
  background-position: center;
  background-repeat: no-repeat;
  position: relative;
  padding: 3em;
  border-top: 15px solid ${HoneyCoDarkGray};
  margin: 0;
`
const StyledTitle = styled.span`
  color: ${HoneyCoDarkGray};
  display: block;
  font-size: 50px;
  text-align: center;
  width: 100%;
`
const FloatingText = styled.p`
  position: absolute;
  top: 20px;
  left: 0;
  font-size: 20px;
  padding: 1em;
  color: white;
  width: 280px;
  background-color: gray;
  text-align: justify;
  letter-spacing: -0.3px;
  line-height: 25px;
  @media(max-width: 1199px){
    display: none;
  }
`
const FloatingTextHeader = styled.span`
  font-weight: bold;
  font-size: 21px;
  display: block;
  text-align: left;
`
const StyledIconsContainer = styled.div`
  height: 500px;
  width: 100%;
  position: relative;
`
const StyledIconContainer = styled.div`
  top: ${(props) => props.top};
  left: ${(props) => props.left};
  position: absolute;
  height: 200px;
  width: 200px;
  border-radius: 50%;
  background-color: rgba(255,255,255, 0.7);
  &:hover{
    background-color: rgba(255,255,255, 1);
    height: 230px;
    width: 230px;
    transition: width 1s height 1s background-color 1s;
  }
`
const IconWrapper = styled.div`
  position: absolute;
  top: calc(50% - 48px);
  left: calc(50% - 50px);
`

const StyledIcon = styled.img`
  text-align: center;
  height: 100px;
  width: 100px;
  vertical-align: bottom;
`

const SmartHome = () =>
  <StyledRow>
    <Container>
      <Row>
        <StyledTitle>A Smart Home is a Safe Home</StyledTitle>
        <FloatingText>
          <FloatingTextHeader>
            HoneyCo homes has you covered.
          </FloatingTextHeader>
          <br/>
            We review hundreds of devices
            to deliver you the best in class
            hardware for you and your home.
            We also install and mantain the
            system so you have a worry-free
            experience.
        </FloatingText>
        <StyledIconsContainer>
          <StyledIconContainer top = '20%' left = '30%'>
            <IconWrapper>
            <StyledIcon src = {Icon1}/>
            </IconWrapper>
          </StyledIconContainer>
          <StyledIconContainer top = '10%' left = '60%'>
            <IconWrapper>
            <StyledIcon src = {Icon2}/>
            </IconWrapper>
          </StyledIconContainer>
          <StyledIconContainer top = '50%' left = '80%'>
            <IconWrapper>
            <StyledIcon src = {Icon3}/>
            </IconWrapper>
          </StyledIconContainer>
          <StyledIconContainer top = '80%' left = '55%'>
            <IconWrapper>
            <StyledIcon src = {Icon4}/>
            </IconWrapper>
          </StyledIconContainer>
          <StyledIconContainer top = '75%' left = '20%'>
            <IconWrapper>
            <StyledIcon src = {Icon5}/>
            </IconWrapper>
          </StyledIconContainer>
        </StyledIconsContainer>
      </Row>
    </Container>
  </StyledRow>

export default SmartHome
