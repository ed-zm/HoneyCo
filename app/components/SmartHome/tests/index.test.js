import React from 'react'
import { shallow } from 'enzyme'
import SmartHome from '../index'

describe('<SmartHome />', () => {
  it('It should render the SmartHome Component', () => {
    const wrapper = shallow(<SmartHome/>)
    expect(wrapper).toMatchSnapshot()
  })
})
