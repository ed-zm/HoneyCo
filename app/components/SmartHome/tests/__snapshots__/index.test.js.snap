// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmartHome /> It should render the SmartHome Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(bt)>
    <gt
        tag="div"
    >
        <bt
            tag="div"
        >
            <styled.span>
                A Smart Home is a Safe Home
            </styled.span>
            <styled.p>
                <styled.span>
                    HoneyCo homes has you covered.
                </styled.span>
                <br />
                We review hundreds of devices to deliver you the best in class hardware for you and your home. We also install and mantain the system so you have a worry-free experience.
            </styled.p>
            <styled.div>
                <styled.div
                    left="30%"
                    top="20%"
                >
                    <styled.div>
                        <styled.img
                            src="IMAGE_MOCK"
                        />
                    </styled.div>
                </styled.div>
                <styled.div
                    left="60%"
                    top="10%"
                >
                    <styled.div>
                        <styled.img
                            src="IMAGE_MOCK"
                        />
                    </styled.div>
                </styled.div>
                <styled.div
                    left="80%"
                    top="50%"
                >
                    <styled.div>
                        <styled.img
                            src="IMAGE_MOCK"
                        />
                    </styled.div>
                </styled.div>
                <styled.div
                    left="55%"
                    top="80%"
                >
                    <styled.div>
                        <styled.img
                            src="IMAGE_MOCK"
                        />
                    </styled.div>
                </styled.div>
                <styled.div
                    left="20%"
                    top="75%"
                >
                    <styled.div>
                        <styled.img
                            src="IMAGE_MOCK"
                        />
                    </styled.div>
                </styled.div>
            </styled.div>
        </bt>
    </gt>
</Styled(bt)>,
  "nodes": Array [
    <Styled(bt)>
      <gt
            tag="div"
      >
            <bt
                  tag="div"
            >
                  <styled.span>
                        A Smart Home is a Safe Home
                  </styled.span>
                  <styled.p>
                        <styled.span>
                              HoneyCo homes has you covered.
                        </styled.span>
                        <br />
                        We review hundreds of devices to deliver you the best in class hardware for you and your home. We also install and mantain the system so you have a worry-free experience.
                  </styled.p>
                  <styled.div>
                        <styled.div
                              left="30%"
                              top="20%"
                        >
                              <styled.div>
                                    <styled.img
                                          src="IMAGE_MOCK"
                                    />
                              </styled.div>
                        </styled.div>
                        <styled.div
                              left="60%"
                              top="10%"
                        >
                              <styled.div>
                                    <styled.img
                                          src="IMAGE_MOCK"
                                    />
                              </styled.div>
                        </styled.div>
                        <styled.div
                              left="80%"
                              top="50%"
                        >
                              <styled.div>
                                    <styled.img
                                          src="IMAGE_MOCK"
                                    />
                              </styled.div>
                        </styled.div>
                        <styled.div
                              left="55%"
                              top="80%"
                        >
                              <styled.div>
                                    <styled.img
                                          src="IMAGE_MOCK"
                                    />
                              </styled.div>
                        </styled.div>
                        <styled.div
                              left="20%"
                              top="75%"
                        >
                              <styled.div>
                                    <styled.img
                                          src="IMAGE_MOCK"
                                    />
                              </styled.div>
                        </styled.div>
                  </styled.div>
            </bt>
      </gt>
</Styled(bt)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SmartHome />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(bt)>
          <gt
                    tag="div"
          >
                    <bt
                              tag="div"
                    >
                              <styled.span>
                                        A Smart Home is a Safe Home
                              </styled.span>
                              <styled.p>
                                        <styled.span>
                                                  HoneyCo homes has you covered.
                                        </styled.span>
                                        <br />
                                        We review hundreds of devices to deliver you the best in class hardware for you and your home. We also install and mantain the system so you have a worry-free experience.
                              </styled.p>
                              <styled.div>
                                        <styled.div
                                                  left="30%"
                                                  top="20%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                                        <styled.div
                                                  left="60%"
                                                  top="10%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                                        <styled.div
                                                  left="80%"
                                                  top="50%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                                        <styled.div
                                                  left="55%"
                                                  top="80%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                                        <styled.div
                                                  left="20%"
                                                  top="75%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                              </styled.div>
                    </bt>
          </gt>
</Styled(bt)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(bt)>
          <gt
                    tag="div"
          >
                    <bt
                              tag="div"
                    >
                              <styled.span>
                                        A Smart Home is a Safe Home
                              </styled.span>
                              <styled.p>
                                        <styled.span>
                                                  HoneyCo homes has you covered.
                                        </styled.span>
                                        <br />
                                        We review hundreds of devices to deliver you the best in class hardware for you and your home. We also install and mantain the system so you have a worry-free experience.
                              </styled.p>
                              <styled.div>
                                        <styled.div
                                                  left="30%"
                                                  top="20%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                                        <styled.div
                                                  left="60%"
                                                  top="10%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                                        <styled.div
                                                  left="80%"
                                                  top="50%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                                        <styled.div
                                                  left="55%"
                                                  top="80%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                                        <styled.div
                                                  left="20%"
                                                  top="75%"
                                        >
                                                  <styled.div>
                                                            <styled.img
                                                                      src="IMAGE_MOCK"
                                                            />
                                                  </styled.div>
                                        </styled.div>
                              </styled.div>
                    </bt>
          </gt>
</Styled(bt)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SmartHome />,
}
`;
